web/eslint.config.js

```diff
@@ -0,0 +1,150 @@
+import eslintPluginPrettierRecommended from "eslint-plugin-prettier/recommended";
+
+// noinspection JSUnusedGlobalSymbols
+export default tseslint.config(
+    eslint.configs.recommended,
+    eslintPluginPrettierRecommended,
+    {
+        ignores: [
+            "**/dist/**",
+            "**/docs/**",
+            "packages/selfhosted/test_assets/swfobject.js",
+        ],
+    },
+    {
+        languageOptions: {
+            ecmaVersion: 2021,
+        },
+        rules: {
+            camelcase: [
+                "error",
+                { properties: "never", allow: ["__webpack_public_path__"] },
+            ],
+            curly: "error",
+            eqeqeq: "error",
+            "no-constructor-return": "error",
+            "no-unused-vars": ["error", { argsIgnorePattern: "^_" }],
+            "prefer-const": "error",
+            "spaced-comment": [
+                "error",
+                "always",
+                { block: { balanced: true }, markers: ["/"] },
+            ],
+        },
+    },
+    {
+        files: ["**/*.ts"],
+        extends: tseslint.configs.recommended,
+        rules: {
+            "@typescript-eslint/no-non-null-assertion": "off",
+            "@typescript-eslint/consistent-type-assertions": [
+                "error",
+                { assertionStyle: "as" },
+            ],
+            "@typescript-eslint/no-unused-vars": [
+                "error",
+                { argsIgnorePattern: "^_" },
+            ],
+            // Disallow const enums, as they can't be used by consumers.
+            // See https://www.typescriptlang.org/docs/handbook/enums.html#const-enum-pitfalls
+            "no-restricted-syntax": [
+                "error",
+                {
+                    selector: "TSEnumDeclaration[const=true]",
+                    message: "Don't declare const enums",
+                },
+            ],
+        },
+    },
+    {
+        files: ["packages/core/src/**", "packages/selfhosted/js/**"],
+        languageOptions: {
+            globals: globals.browser,
+        },
+    },
+    {
+        files: ["packages/extension/src/**"],
+        languageOptions: {
+            globals: { ...globals.browser, ...globals.webextensions },
+        },
+    },
+    {
+        files: ["packages/core/tools/**"],
+        languageOptions: {
+            globals: globals.node,
+            ecmaVersion: 2022, // Needed for top-level-await.
+        },
+    },
+    {
+        files: ["packages/**/webpack.config.js"],
+        languageOptions: {
+            globals: globals.node,
+        },
+    },
+    {
+        files: ["packages/core/test/**"],
+        languageOptions: {
+            globals: globals.mocha,
+        },
+    },
+    {
+        files: ["packages/selfhosted/test/**"],
+        languageOptions: {
+            globals: {
+                ...globals.mocha,
+                ...globals.node,
+                ...globals.mocha,
```

Comment:
You probably meant `browser`, `node`, `mocha` here, instead of `mocha` twice.

Reply:
Interesting that it worked, huh!

Reply:
Turns out that we don't need it because typescript. This affects js. Removed!